### PR TITLE
Migration infra

### PR DESCRIPTION
**Pull request by suhanishri**
1. This PR contains migration infra containing logger file. 
2. Output tables for sapHana, sapNetweaver and unsupported provider type
3. Code is added for converting ARM id of AMSv1 to extract subscriptionId, resource grp and resource name
4. Code added to fetch secrets from keyVault
5. Customer is need to input ARM id of AMSv1 and providerType they wish to migrate. If providerType flag is not provided then automatically all supported providers are migrated
3. This PR has entry point from Migration.ps1
4. Some other parameters are added for testing. We can remove that later
5. The PR also contains a sample MigrationLog file

### CODE REVIEW

File: MigrationScript/MigrationLog.txt [hunk 1]
```diff
@@ -0,0 +1,98 @@
+﻿**********************
+Windows PowerShell transcript start
```
Comment (amitms2103) [inline MigrationScript/MigrationLog.txt]: Do we need to add this file in the PR ?

File: MigrationScript/MigrationLog2.txt [hunk 2]
```diff
@@ -0,0 +1,81 @@
+﻿**********************
+Windows PowerShell transcript start
+Start time: 20220217172519
+Username: FAREAST\sshrivastava
```
Comment (amitms2103) [inline MigrationScript/MigrationLog2.txt]: Same here is this file needed ?

**Comment by amitms2103** [comment]
Suhani consider restructuring the code a bit more to make the directory structure a bit more cleaner.

File: MigrationScript/MigrationLog2.txt (hunk 2 above)
Comment (suhanishri) [reply MigrationScript/MigrationLog2.txt]: Removed this file

File: MigrationScript/MigrationLog.txt (hunk 1 above)
Comment (suhanishri) [reply MigrationScript/MigrationLog.txt]: Just added a sample for log. Shall i remove it?

Comment (amitms2103) [reply MigrationScript/MigrationLog.txt]: I think better remove it as it contains subscriptionId etc details and repo will be public. We can provide customers some samples if they need it.

**Comment by suhanishri** [comment]
> Suhani consider restructuring the code a bit more to make the directory structure a bit more cleaner.

Yes agreed. I've restructured the code. Let me know if any further modularization required

File: MigrationScript/MigrationLog.txt (hunk 1 above)
Comment (suhanishri) [reply MigrationScript/MigrationLog.txt]: Okay removed it

File: MigrationScript/ConsoleLogger.ps1 [hunk 3]
```diff
@@ -0,0 +1,41 @@
+﻿Class ConsoleLogger
```
Comment (amitms2103) [inline MigrationScript/ConsoleLogger.ps1:1]: Looks Good

**Review by amitms2103 — APPROVED** [review]
